### PR TITLE
Honor environment variable to set gRPC listener address

### LINKED ISSUE #383: Introduce environment variable for Refinery's gRPC listener's bind address
**Is your feature request related to a problem? Please describe.**

When choosing whether to serve Refinery behind a proxy terminating TLS, sometimes we want to bind Refinery's gRPC listener to all network interfaces (specified via the IP address _0.0.0.0_) in order to allow unencrypted calls from clients anywhere, but sometimes we want to bind this listener to only the loopback interface (via address _127.0.0.1_).

At present, we can only configure the gRPC listener's bind address via the Refinery configuration file. When operating with or without TLS in front of Refinery, this is the only detail that has to change in the configuration file. Much like with [the "REFINERY_REDIS_HOST" environment variable](https://github.com/honeycombio/refinery/blob/5d57c7468ded2bbb5c4b95868b9d2a6befb22539/config/file_config.go#L97), we would like to introduce an environment variable to set the gRPC listener's bind address, setting [the `configContents.GRPCListenAddr` field](https://github.com/honeycombio/refinery/blob/423f235c88de424acc34e2a1fc822d878254973c/config/file_config.go#L31) from afar. Doing so would allow us to use a common configuration file, and vary only this environment variable in our Kubernetes pod spec for Refinery.

**Describe the solution you'd like**

Introduce a new environment variable consumed by Refinery, named something like "REFINERY_GPRC_LISTEN_ADDR" for parity with the similar environment variable names. As with other Viper-managed configuration, [setting this environment variable would take precedence over a value set in the configuration file](https://github.com/spf13/viper#why-viper).

I expect the implementation would look something like the following in the `config.NewConfig` function:
```go
c.BindEnv("GRPCListenAddr", "REFINERY_GPRC_LISTEN_ADDR")
```

**Describe alternatives you've considered**

We can generate different configuration files for each of these two scenarios, but doing so introduces more complexity into our Kubernetes manifest preparation system, and we wind up with two nearly identical configurations in play, without hoisting out the lone detail that varies to make the intended similarity and difference clear.

**Additional context**

Viper also offers [its `AutomaticEnv` facility](https://github.com/spf13/viper/#working-with-environment-variables), opening **all** the configuration fields to access via environment variables, albeit with some clumsy names. [I mentioned this possibility in the "discuss-refinery" channel](https://honeycombpollinators.slack.com/archives/C01BZ3J7RMJ/p1643061358024000) of the "Honeycomb Pollinators" Slack workspace.

### PR DESCRIPTION
**Pull request by seh**
Introduce the "REFINERY_GRPC_LISTEN_ADDRESS" environment variable to allow setting Refinery's `GRPCListenAddr` configuration field, overriding any such configuration read from a file.

Fixes #383.

### CODE REVIEW

File: config/config_test.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// +build all race
+//go:build all || race
```
Comment (MikeGoldsmith) [inline config/config_test.go:1]: I'd prefer to update all go build tags in the project at the same time. We can do that in a separate PR 👍🏻

**Review by MikeGoldsmith — APPROVED** [review]
Thanks for the PR @seh - it looks great. Just one small ask on reverting the build tags.

It helps address some of #275 too.

File: config/config_test.go (hunk 1 above)
Comment (seh) [reply config/config_test.go]: Please see my most recent commit f65b90c64a790eafb64ab2fe47f6742abbe7b164 in which I reverted that change. _go fmt_ isn't happy about it, but it will come around.

Comment (MikeGoldsmith) [reply config/config_test.go]: Thanks - I agree we need to move the newer standard and will do the change package wide separately in another PR 👍🏻
